src/plugins/intel_gna/gna_model_serial.cpp

```diff
@@ -558,6 +558,11 @@ void GNAModelSerial::ImportInputs(std::istream &is, void* basePtr, GNAPluginNS::
         for (uint8_t tId = 0; tId < ep.tensor_names_count; ++tId) {
             input.tensor_names.insert(readString(is));
         }
+
+        // To support importing of <= 2.7 format where tensor_names were not present
+        if (model_header_.version.major == 2 && model_header_.version.minor <= 7 && ep.tensor_names_count == 0) {
```

Comment:
1. you use the same condition  in two places in the code. It would be better to create one method which would check condition 
 e.g.: bool isModelImportSupportedWhenTensorNamesAreNotPresent(version, count){} -
in the future you would need to change only one place
3. I would also declare 2 and 7 as static constexpr to not use magic numbers if possible.

Reply:
Applied,
@marcinkusm please verify